test/PoolManagerDelegateCall.t.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+import {Test} from "forge-std/Test.sol";
+import {IHooks} from "../src/interfaces/IHooks.sol";
+import {IPoolManager} from "../src/interfaces/IPoolManager.sol";
+import {NoDelegateCall} from "../src/NoDelegateCall.sol";
+import {PoolManager} from "../src/PoolManager.sol";
+import {PoolSwapTest} from "../src/test/PoolSwapTest.sol";
+import {ProxyPoolManager} from "../src/test/ProxyPoolManager.sol";
+import {Deployers} from "./utils/Deployers.sol";
+
+contract PoolManagerDelegateCall is Test, Deployers {
+    // override to use ProxyPoolManager
+    function deployFreshManager() internal virtual override {
```

Comment:
Could you add these tests to NoDelegateCall.t.sol?

Reply:
Thanks for the feedback, will have these changes in shortly.

Want to pick your brain on which functions to have noDelegateCall though, now that it's logically uncoupled from unlock. I see two potential approaches:

- protect just the core pool logic, namely modifyLiquidity, swap, and donate; and leave it off mint/burn and settle/take
- protect all interface functions, that would also include sync and updateDynamicLFFee

Somehow I feel the line currently drawn somewhat arbitrary...

Reply:
Changes re comment checked in

Reply:
forge fmt made difference though, not sure what the formatting diffs may be..

Reply:
I do agree that the line is somewhat arbitrary currently. I think core functionality swap/modify/donate makes sense. I'd probably opt for it to also be on initialize. But, I don't feel strongly it needs to be on mint/burn and take/settle though. @saucepoint / @gretzke thoughts?

Reply:
for reference this was what we previously had changed https://github.com/Uniswap/v4-core/pull/549/files#diff-84227e9327ad53c4ad37657de3ed4ee30143b70e038ddabd7e6214eba4f0865c

Reply:
Ok I think my final/full opinion here re:exact functions for this modifier. We should include it on:

- initialize
- modifyLiquidity
- swap
- donate

Justification: Want to keep this overhead minimal & those are the core swap lifecycle functions. I don't see any harm with someone delegatecalling mint/burn and take/settle sync/ other public/external functions.

Let me know thoughts @jeffishjeff